### ISSUE TITLE
An error after training for a while

### ISSUE THREAD

**Issue by Luoyu-Wang** [MIC-DKFZ/nnUNet#2866]
Hi, I encountered a problem where it exited with an error after training for a while. Could you tell me how to resolve it?

OS: Ubuntu 22.04

GPU: NVIDIA RTX 3060 (12GB VRAM)

Driver: NVIDIA 570

CUDA: 11.8

Installation:

Fresh Conda environment

Latest PyTorch and nnU-Net per documentation

Commend:

nnUNetv2_plan_experiment -d 101 -gpu_memory_target 11 -pl nnUNetPlannerResEncM

nnUNetv2_train 101 3d_fullres 0  --npz  (errer!)

Errors:

Fatal Python error: none_dealloc: deallocating None
Python runtime state: initialized

Thread 0x000075acd3e00640 (most recent call first):
  <no Python frame>

Thread 0x000075acd7e00640 (most recent call first):
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 324 in wait
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 607 in wait
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/tqdm/_monitor.py", line 60 in run
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 973 in _bootstrap

Thread 0x000075aced600640 (most recent call first):
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/multiprocessing/popen_fork.py", line 27 in poll
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/multiprocessing/process.py", line 165 in is_alive
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/batchgenerators/dataloading/nondet_multi_threaded_augmenter.py", line 101 in <listcomp>
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/batchgenerators/dataloading/nondet_multi_threaded_augmenter.py", line 101 in results_loop
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 953 in run
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 973 in _bootstrap

Thread 0x000075acfaa00640 (most recent call first):
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/multiprocessing/popen_fork.py", line 27 in poll
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/multiprocessing/process.py", line 165 in is_alive
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/batchgenerators/dataloading/nondet_multi_threaded_augmenter.py", line 101 in <listcomp>
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/batchgenerators/dataloading/nondet_multi_threaded_augmenter.py", line 101 in results_loop
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 953 in run
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 973 in _bootstrap

Thread 0x000075ad35e00640 (most recent call first):
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/torch/_inductor/compile_worker/subproc_pool.py", line 55 in _recv_msg
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/torch/_inductor/compile_worker/subproc_pool.py", line 191 in _read_thread
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 953 in run
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/threading.py", line 973 in _bootstrap

Current thread 0x000075aeb4dea740 (most recent call first):
  Garbage-collecting
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/_api/__init__.py", line 125 in check_in_list
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/cbook.py", line 356 in process
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/artist.py", line 425 in pchanged
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/artist.py", line 1065 in set_visible
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/artist.py", line 1209 in _update_props
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/artist.py", line 1223 in update
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/text.py", line 205 in update
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/text.py", line 155 in __init__
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axis.py", line 176 in __init__
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axis.py", line 368 in __init__
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axis.py", line 1592 in _get_tick
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axis.py", line 1664 in get_major_ticks
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axis.py", line 1283 in _update_ticks
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axis.py", line 1351 in get_tightbbox
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/artist.py", line 1402 in _get_tightbbox_for_layout_only
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/axes/_base.py", line 4554 in get_tightbbox
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/artist.py", line 1402 in _get_tightbbox_for_layout_only
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/_tight_layout.py", line 82 in _auto_adjust_subplotpars
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/_tight_layout.py", line 266 in get_tight_layout_figure
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/layout_engine.py", line 188 in execute
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/figure.py", line 3640 in tight_layout
  File "/home/wly/anaconda3/envs/nnunetv2/lib/python3.10/site-packages/matplotlib/pyplot.py", line 2844 in tight_layout
  File "/home/wly/nnUNet/nnunetv2/training/logging/nnunet_logger.py", line 94 in plot_progress_png
  File "/home/wly/nnUNet/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py", line 1145 in on_epoch_end
  File "/home/wly/nnUNet/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py", line 1381 in run_training
  File "/home/wly/nnUNet/nnunetv2/run/run_training.py", line 207 in run_training
  File "/home/wly/nnUNet/nnunetv2/run/run_training.py", line 266 in run_training_entry
  File "/home/wly/anaconda3/envs/nnunetv2/bin/nnUNetv2_train", line 8 in <module>

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, torch._C, torch._C._dynamo.autograd_compiler, torch._C._dynamo.eval_frame, torch._C._dynamo.guards, torch._C._dynamo.utils, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, torch._C._special, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy._lib._ccallback_c, charset_normalizer.md, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.special._ellip_harm_2, _ni_label, scipy.ndimage._ni_label, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.spatial.transform._rotation, skimage.morphology._misc_cy, skimage.measure._ccomp, _skeletonize_lee_cy, skimage.morphology._skeletonize_lee_cy, skimage.morphology._skeletonize_various_cy, skimage._shared.geometry, skimage.measure._pnpoly, skimage.morphology._convex_hull, skimage.morphology._grayreconstruct, skimage.morphology._extrema_cy, skimage.morphology._flood_fill_cy, skimage.morphology._max_tree, skimage.transform._warps_cy, cc3d.fastcc3d, SimpleITK._SimpleITK, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, numexpr.interpreter, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, PIL._imaging, msgpack._cmsgpack, blosc2.blosc2_ext, requests.packages.charset_normalizer.md, requests.packages.chardet.md, ndindex._slice, ndindex._tuple, kiwisolver._cext, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize._cython_nnls, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.integrate._lsoda, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.interpolate._bspl, scipy.special.cython_special, scipy.stats._stats, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._biasedurn, scipy.stats._stats_pythran, scipy.stats._levy_stable.levyst, scipy.stats._ansari_swilk_statistics, scipy.stats._mvn, scipy.stats._rcont.rcont, scipy.cluster._vq, scipy.cluster._hierarchy, scipy.cluster._optimal_leaf_ordering, sklearn.__check_build._check_build, sklearn.utils._isfinite, sklearn.utils.sparsefuncs_fast, sklearn.utils.murmurhash, sklearn.utils._openmp_helpers, sklearn.metrics.cluster._expected_mutual_info_fast, sklearn.preprocessing._csr_polynomial_expansion, sklearn.preprocessing._target_encoder_fast, sklearn.metrics._dist_metrics, sklearn.metrics._pairwise_distances_reduction._datasets_pair, sklearn.utils._cython_blas, sklearn.metrics._pairwise_distances_reduction._base, sklearn.metrics._pairwise_distances_reduction._middle_term_computer, sklearn.utils._heap, sklearn.utils._sorting, sklearn.metrics._pairwise_distances_reduction._argkmin, sklearn.metrics._pairwise_distances_reduction._argkmin_classmode, sklearn.utils._vector_sentinel, sklearn.metrics._pairwise_distances_reduction._radius_neighbors, sklearn.metrics._pairwise_distances_reduction._radius_neighbors_classmode, sklearn.metrics._pairwise_fast, sklearn.utils._random, markupsafe._speedups, cuda_utils, __triton_launcher, PIL._imagingmath (total: 199)

**Comment by llevitis** [comment]
Hi @Luoyu-Wang - did you manage to solve this problem? I'm running into the same issue, also with a recently created python 3.10 conda environment and the latest PyTorch and nnUNet installation. Thank you!!
